Add package liblouis version 3.24.0

suresh-thelkar opened the pull request:
<!--
COMMENT BLOCKS WILL NOT BE INCLUDED IN THE PR.
Feel free to delete sections of the template which do not apply to your PR, or add additional details
-->

###### Merge Checklist  <!-- REQUIRED -->
<!-- You can set them now ([x]) or set them later using the Github UI -->
**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary <!-- REQUIRED -->
<!-- Quick explanation of the changes. -->
Add package liblouis version 3.22.0

###### Change Log  <!-- REQUIRED -->
<!-- Detail the changes made here. -->
<!-- Please list any packages which will be affected by this change, if applicable. -->
<!-- Please list any CVES fixed by this change, if applicable. -->
- NA

###### Does this affect the toolchain?  <!-- REQUIRED -->
<!-- Any packages which are included in the toolchain should be carefully considered. Make sure the toolchain builds with these changes if so. -->
<!-- Update: manifests/package/toolchain_*.txt, pkggen_core_*.txt, update_manifests.sh -->
<!-- To validate: make clean; make workplan REBUILD_TOOLCHAIN=y DISABLE_UPSTREAM_REPOS=y CONFIG_FILE="" ... -->
**NO**

###### Associated issues  <!-- optional -->
<!-- Link to Github issues if possible. -->
<!-- you can use "fixes #xxxx" to auto close an associated issue once the PR is merged -->
- NA

###### Links to CVEs  <!-- optional -->
- NA
###### Test Methodology
<!-- How was this test validated? i.e. local build, pipeline build etc. -->
Buddy Build
- [AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=290720&view=results)
- [ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=290719&view=results)

suresh-thelkar commented:
Please note that all linter warnings are fixed except a few false warnings.

oliviacrain commented on SPECS-EXTENDED/liblouis/liblouis.spec:
```diff
@@ -0,0 +1,407 @@
+Name:           liblouis
+Version:        3.22.0
+Release:        2%{?dist}
+License:        LGPLv3+
```
Small nit: We're trying to slowly change these license tags to use [SPDX expressions](https://spdx.org/licenses/). Mind using the SPDX name for this and other licenses?

```suggestion
License:        LGPL-3.0-or-later
```

suresh-thelkar replied:
Thanks for suggesting this. I have modified the license tags as per SPDX expressions

oliviacrain commented on SPECS-EXTENDED/liblouis/liblouis.spec:
```diff
@@ -0,0 +1,407 @@
+
+%package        devel
+Summary:        Development files for %{name}
+License:        LGPLv3+
```
```suggestion
License:        LGPL-3.0-or-later
```

oliviacrain commented on SPECS-EXTENDED/liblouis/liblouis.spec:
```diff
@@ -0,0 +1,407 @@
+
+%package        utils
+Summary:        Command-line utilities to test %{name}
+License:        GPLv3+
```
```suggestion
License:        GPL-3.0-or-later
```

oliviacrain commented on SPECS-EXTENDED/liblouis/liblouis.spec:
```diff
@@ -0,0 +1,407 @@
+%package -n python3-louis
+%{?python_provide:%python_provide python3-louis}
+Summary:        Python 3 language bindings for %{name}
+License:        LGPLv3+
```
```suggestion
License:        LGPL-3.0-or-later
```

oliviacrain commented on SPECS-EXTENDED/liblouis/liblouis.spec:
```diff
@@ -0,0 +1,407 @@
+License:        LGPLv3+
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+URL:            http://liblouis.org
```
Can we change this to point somewhere else? Judging by the registration history of the URL, they're in the middle of moving domain registrars and this currently leads to an ad-infested registrar page.

suresh-thelkar replied:
I have modified the URL to the following. Hope that is fine. 
https://github.com/liblouis/liblouis

oliviacrain commented on SPECS-EXTENDED/liblouis/liblouis.spec:
```diff
@@ -0,0 +1,407 @@
+This package provides the documentation for liblouis.
+%endif
+
+%global __os_install_post %{nil}
+%global debug_package %{nil}
```
Can we document the rationale for these flags?

suresh-thelkar replied:
I realized that these two macros are not needed and removed them from the latest commit. Please take a look.

oliviacrain commented on SPECS-EXTENDED/liblouis/liblouis.spec:
```diff
@@ -0,0 +1,407 @@
+
+install -d %{buildroot}%{python3_sitelib}/louis
+install -pm 0644 __init__.py %{buildroot}%{python3_sitelib}/louis/
+%py_byte_compile %{__python3} %{buildroot}%{python3_sitelib}/louis/
```
I'm not sure I understand what's happening here- why are we disabling automatic compilation as the beginning of the spec, and then doing it manually here?

suresh-thelkar replied:
I have analyzed further on this. It seems the following guideline is used by the people from where we imported our spec file initially for the various reasons.

"One MUST disable their automatic compilation, and compile them explicitly by the %py_byte_compile macro if and only if the bytecompilation is desired."

But when it comes to Mariner I believe we no need to follow the above guideline and hence I removed the following code. 
 `%py_byte_compile %{__python3} %{buildroot}%{python3_sitelib}/louis/`

After successful build, I have observed that In our case the respective *.py files are automatically byte compiled which is what we wanted.  
`rpm -qlp RPMS/noarch/python3-louis-3.24.0-2.cm2.noarch.rpm
/usr/lib/python3.9/site-packages/louis
/usr/lib/python3.9/site-packages/louis/__init__.py
/usr/lib/python3.9/site-packages/louis/__pycache__
/usr/lib/python3.9/site-packages/louis/__pycache__/__init__.cpython-39.opt-1.pyc
/usr/lib/python3.9/site-packages/louis/__pycache__/__init__.cpython-39.pyc
`

suresh-thelkar commented on SPECS-EXTENDED/liblouis/liblouis.spec:
```diff
@@ -0,0 +1,407 @@
+# Turn off the brp-python-bytecompile script
+%global __os_install_post %(echo '%{__os_install_post}' | sed -e 's!/usr/lib[^[:space:]]*/brp-python-bytecompile[[:space:]].*$!!g')
```
Yes you are right. In fact there is no need to disable automatic byte compilation and hence I removed the above portion of the code. Please look at the latest commit.

suresh-thelkar commented:
@oliviacrain, Thank for the review. I have addressed all your comments. Please take a look at the latest commit/s and let me know if you have any further comments. 

I have also run Buddy builds after implementing the review comments. Both of them passed. 
[AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=291997&view=results)
[ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=291998&view=results)

suresh-thelkar commented:
I have once again ran Buddy Builds after fixing linter warnings and all the commits. Both the following builds are passing. 
[AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=292006&view=results)
[ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=292007&view=results)

oliviacrain reviewed (approved):
Looks great!

suresh-thelkar commented:
> Looks great!

Thank you for the approval.

suresh-thelkar commented:
I have once again ran buddy builds after rebasing. Here are the results. Both of them passed.
- [AMD64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=293966&view=results)
- [ARM64](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=293954&view=results)
